config/errors_block.go

```diff
@@ -29,146 +29,167 @@ type IgnoreBlock struct {
 	Signals         map[string]cty.Value `cty:"signals" hcl:"signals,optional"`
 }
 
-// Clone creates a deep copy of ErrorsConfig
+// Clone returns a deep copy of ErrorsConfig
 func (c *ErrorsConfig) Clone() *ErrorsConfig {
 	if c == nil {
 		return nil
 	}
 
-	clone := &ErrorsConfig{
-		Retry:  make([]*RetryBlock, len(c.Retry)),
-		Ignore: make([]*IgnoreBlock, len(c.Ignore)),
+	return &ErrorsConfig{
+		Retry:  cloneRetryBlocks(c.Retry),
+		Ignore: cloneIgnoreBlocks(c.Ignore),
 	}
+}
 
-	// Clone Retry blocks
-	for i, retry := range c.Retry {
-		clone.Retry[i] = retry.Clone()
+// Merge combines the current ErrorsConfig with another one, prioritizing the other config
+func (c *ErrorsConfig) Merge(other *ErrorsConfig) {
+	if c == nil || other == nil {
```

Comment:
If you merge a config into a nil config, shouldn't you expect that `c` becomes `other`?

Reply:
Since changes to the receiver don't affect the caller, initializing c has no outside effect - better to return early if null.